backend/src/main/java/com/benine/backend/camera/ipcameracontrol/IPCamera2.java

```diff
@@ -0,0 +1,175 @@
+
+  private int id = -1;
+
+  /**
+   *  Create a new IP Camera object.
+   *  @param ip address of this camera.
+   */
+  public IPCamera2(String ip) {
+    ipaddress = ip;
+  }
+
+  @Override
+  public String sendCommand(String command) {
+    try {
+      InputStream com = new URL("http://" + ipaddress + "/cgi-bin/CGIProxy.fcgi?cmd=" + command + "&usr=user&pwd=geheim").openStream();
+      DocumentBuilderFactory dbFactory  = DocumentBuilderFactory.newInstance();
+      DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+      Document doc = dBuilder.parse(com);
+      doc.getDocumentElement().normalize();
+      NodeList l = doc.getElementsByTagName("result");
+      if (Integer.parseUnsignedInt(l.item(0).getTextContent()) != 0) {
+        throw new IOException();
+      }
+      return doc.toString();
+    } catch (IOException e) {
+      e.printStackTrace();
+    } catch (SAXException e){
+      e.printStackTrace();
+    } catch (ParserConfigurationException e){
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public int getZoomPosition() throws CameraConnectionException {
+    //Not supported
+    return 50;
+  }
+
+  @Override
+  public void zoomTo(int zpos) throws CameraConnectionException {
+    if (zpos > 50) {
+      sendCommand("zoomIn");
+    } else if (zpos < 50) {
+      sendCommand("zoomOut");
+    } else {
+      sendCommand("zoomStop");
+    }
+  }
+
+  @Override
+  public void zoom(int dir) throws CameraConnectionException {
+    int speed = Math.abs(dir - 50);
+    if (speed < 17) {
+      sendCommand("setZoomSpeed&speed=2");
+    } else if (speed < 33) {
+      sendCommand("setZoomSpeed&speed=1");
+    } else {
+      sendCommand("setZoomSpeed&speed=0");
+    }
+    if (dir > 50) {
+      sendCommand("zoomIn");
+    } else if (dir < 50) {
+      sendCommand("zoomOut");
+    } else {
+      sendCommand("zoomStop");
+    }
+  }
+
+  @Override
+  public void moveTo(Position pos, int panSpeed, int tiltSpeed)
+      throws CameraConnectionException {
+    //not supported
+  }
+
+  @Override
+  public void move(int pan, int tilt) throws CameraConnectionException {
+    int speed = pan;
+    speed = Math.abs(speed - 50);
+    if (speed < 10) {
+      sendCommand("setPTZSpeed&speed=4");
+    } else if (speed < 20) {
+      sendCommand("setPTZSpeed&speed=3");
+    } else if (speed < 30) {
+      sendCommand("setPTZSpeed&speed=2");
+    } else if (speed < 40) {
+      sendCommand("setPTZSpeed&speed=1");
+    } else {
+      sendCommand("setPTZSpeed&speed=0");
+    }
+    if (pan == 50 && tilt == 50) {
+      sendCommand("ptzStopRun");
+    } else if (pan > 85) {
+      sendCommand("ptzMoveRight");
+    } else if (pan < 15) {
+      sendCommand("ptzMoveLeft");
+    } else if (tilt > 85) { 
+      sendCommand("ptzMoveUp");
+    } else if (tilt < 15) { 
+      sendCommand("ptzMoveDown");
+    } 
+    
+    
+      
+  }
+
+  @Override
+  public Position getPosition() throws CameraConnectionException {
```

Comment:
It would be better to do something here to indicate the action is not supported.